packages/graphql/letter-to-santa/optionality.md

```diff
@@ -0,0 +1,1485 @@
+
+**While we have a way to tell "default required" emitters that a property is _optional_, we do not have a way to tell "default optional" emitters that a property is _required_.**
+
+### Contextual Optionality/Requiredness
+
+The second problem is that optionality and requiredness are defined at the model level, but they may need to change based on the context in which the model is used.
+
+<a name="patch"></a>
+The primary existing example of this is the way that `@typespec/http` handles `PATCH` operations.
+
+When the `@patch` decorator is applied to an operation, a [new "view" of that model](https://typespec.io/docs/language-basics/visibility/#_top) is created with two primary behaviors:
+
+1. Only properties with the `update` visibility are included in the model. This is consistent with the rules of [automatic visibility](https://typespec.io/docs/libraries/http/operations/#automatic-visibility), and is roughly equivalent to applying the `@withLifecycleUpdate` decorator (or `@withVisibility(Lifecycle.update)`) to a model to create the update "view".
+2. All properties are made optional. This behavior is [implicit based on the visibility of the properties](https://github.com/microsoft/typespec/pull/1345), and is not configurable.
+
+The first behavior is not uncommon; indeed, the `Http` library defines [default visibility modifiers for all HTTP verbs][http-default-verb-visibility].
+Since these are merely default modifiers, the TypeSpec developer can override them by applying the `@parameterVisibility` and/or `@returnTypeVisibility` decorator to the operation.
+
+The latter is unique to `PATCH`. It has been [discussed previously](https://github.com/microsoft/typespec/issues/2150#issuecomment-1622215786) that perhaps this behavior should also be overridable with a decorator:
+
+> Add some decorator to explicitly say that the input properties are made optional or not. This can be combined with (1) or (2) or status quo as an override.
+ 
+As recently as Feb 4, 2025, we're still looking for a way to [make this behavior more explicit](https://github.com/microsoft/typespec/discussions/5759#discussioncomment-12057390).
+
+**There is no current way for a TypeSpec developer to specify optionality/requiredness based on context.**
+
+<br>
+
+# Implementation
+
+<a name="terminology"></a>
+## Terminology
+
+To speak more clearly about the concepts of visibility and requiredness, we will refer to the following in the rest of the document.
+
+1. <a name="context-modifier"></a>"visibility modifier" is called "context modifier"
+2. "visibility class" is called "context class"
+3. (possibly) "visibility" will be renamed "contextual visibility", mirroring "contextual requiredness".
+
+## `!` symbol
+
+The `!` symbol is used to mark a property as _explicitly required_.
+
+It is specified in the same manner as the `?` symbol to [mark a property as optional][optional-properties].
+
+```typespec
+model Dog {
+  address!: string;
```

Comment:
We should probably mention that this has no effect in openapi i.e.
```
model Dog {
   address: string;
}
```

is equivalent to

```
model Dog {
     address!: string;
}
```

Reply:
Good point; I may also want to say that
```typespec
model Dog {
   address!: string | null;
}
```
is a perfectly valid construction.